ocaml/libs/tracing/tracing.ml

```diff
@@ -15,13 +15,65 @@ module D = Debug.Make (struct let name = "tracing" end)
 
 open D
 
+module W3CBaggage = struct
+  module Key = struct
+    let is_valid_key str =
+      let is_tchar = function
+        | '0' .. '9'
+        | 'a' .. 'z'
+        | 'A' .. 'Z'
+        | '!'
+        | '#'
+        | '$'
+        | '%'
+        | '&'
+        | '\''
+        | '*'
+        | '+'
+        | '-'
+        | '.'
+        | '^'
+        | '_'
+        | '`'
+        | '|'
+        | '~' ->
+            true
+        | _ ->
+            false
+      in
+      String.to_seq str |> Seq.for_all (fun c -> is_tchar c)
+  end
+
+  module Value = struct
+    type t = string
+
+    let serialise str =
+      let encode_char x =
+        match x with
+        (* Encode anything that isn't in basic US-ASCII or is a Control, whitespace, DQUOTE , ; or \ *)
+        | '\000' .. '\032' | '"' | ',' | ';' | '\\' | '\128' .. '\255' ->
+            Printf.sprintf "%%%02X" (Char.code x)
+        | _ ->
+            String.make 1 x
+      in
+      String.to_seq str
```

Comment:
If this is used a lot, we should use something more efficient: 
* first count the number of chars that need to be escaped by iterating over it
* if zero - do nothing
* if non-zero: create a string of the right size and copy from old to new while escaping those that need it. An escaped char has a defined length - so it is possible to know the required space.
* if this is used rarely or the chance of hitting an escape is low, the escaping mechanism could still use something similar to what you are doing now but it creates a lot of garbage.

Reply:
I think both the chance of hitting an escape is low (because these are user input characters and users wouldn't typically use " , ; \ in an attribute value) and it is rarely used because it is only executed when an observer parameter is changed, which wouldn't be very often. So I think it is fine as is for now :)

Reply:
But this code is doing a lot of work even when there is nothing to do: for a string of length _n_ it creates _n_ (new) strings of length 1 and later concatenates them (into a new string) when it could have returned the original string. And this being part of tracing, I would not assume that this function is called infrequently.

Reply:
Yes that is rather inefficient, I'll take a look at it when I can. Though whilst it's part of tracing, it isn't used as frequently as traces, it's only used when someone changes an Observer's attributes on the commandline